### ISSUE TITLE
[Feature] Fighters auto-dock to replenish ammo/shields/power, then relaunch

### ISSUE THREAD

**Issue by Tadrix** [endless-sky/endless-sky#2336]
What is says on the tin. In Endless Sky it would be very tedious to micromanage carrier fleet, thus I propose this:

If fighter has no more ammo (it can't shoot at all), it docks. If its carrier is carrying more ammo (in cargo), it takes some time to reload (recharging everything else during this time) and then relaunch; otherwise, remain docked.
If fighter is running on batteries, on low power it docks to recharge (recharging shields and reloading ammo, if possible, in the process), then relaunches.
If fighter has <25% shields left, it docks to recharge them (charging batteries and reloading ammo, if possible, in the mean time), then relaunches.
Carrier could also make repairs to fighter hulls?

This would make carriers much more useful, and also there would be more sense to use battery-powered ships.
Some sort of kamikaze order to fighters would be also nice, but it would probably warrant separate issue.

**Comment by Wrzlprnft** [comment]
i think this came up before, but stayed unresolved.
Though, that would probably require fighters memorizing to which carrier they belong, and i recall that not being the case.

**Comment by zwparchman** [comment]
One of the features of my #1971 pull requests hits most of these features. Hulls repair if the carrier has a hull regeneration module in the base game already. Battery management is not included in my PR, but it would be trivial to add.

**Comment by Tadrix** [comment]
@Wrzlprnft If it's not the case, they could just retreat to a nearest carrier with free bays.

**Comment by jafdy** [comment]
@Wrzlprnft it is the case so long as the carrier stays in-system. If you leave the system and come back you get the fighter mass moving from one carrier to the other for a while until they are all docked.

**Comment by Amazinite** [comment]
Is the only request from this issue that isn't done the restocking ammo part?

**Comment by warp-core** [comment]
Did that PR actually cover docking for power?

**Comment by tehhowch** [comment]
> Did that PR actually cover docking for power?

No, docking to replenish power was not an intended use case to be satisfied by that PR.

**Comment by warp-core** [comment]
This issue seems to be requesting docking for power as well. I think there is a PR that does that. So, unless we just don't want that functionality, I think this issue should be reopened?

**Comment by EjoThims** [comment]
Reopened to link to https://github.com/endless-sky/endless-sky/pull/6726 covering the last requested piece.

**Comment by samrocketman** [comment]
I updated `Related issues` in 6726 to note that it will close this issue when it is merged.
